src/main/native/org/apache/commons/crypto/cipher/OpenSslNative.c

```diff
@@ -716,3 +677,52 @@ JNIEXPORT void JNICALL Java_org_apache_commons_crypto_cipher_OpenSslNative_clean
   EVP_CTX_Wrapper *wrapper = CTX_WRAPPER(ctx);
   free_context_wrapper(wrapper);
 }
+
+static int check_update_max_output_len(JNIEnv *env, EVP_CIPHER_CTX *context, jlong ctx, int input_len, int max_output_len)
```

Comment:
All these arguments feel a little redundant. e.g. you don't need `context` since you can just use `wrapper->context`.

With some small changes you could also just pass the wrapper directly from the callers, instead of passing the jni env + the pointer address.

Reply:
I replaced the env and address arguments with the wrapper argument in functions that required the wrapper.